substrate/client/network/sync/src/strategy/chain_sync.rs

```diff
@@ -1665,7 +1667,13 @@ where
 					None
 				}
 			})
-			.collect()
+			.collect::<Vec<_>>();
+
+		if !requests.is_empty() {
```

Comment:
nit: I would add a comment around this racing with `peer_block_request` to not forget about this in the future

Reply:
addressed in https://github.com/paritytech/polkadot-sdk/pull/5774